packages/dapp-svelte-wallet/api/src/lib-wallet.js

```diff
@@ -244,46 +244,49 @@ export async function makeWallet({
     return proposal;
   };
 
-  async function updateInboxState(id, offer) {
+  async function updateInboxState(id, offer, doPush = true) {
     // Only sent the uncompiled offer to the client.
-    const {
-      instanceHandleBoardId,
-      installationHandleBoardId,
-      proposalTemplate,
-    } = offer;
+    const { proposalTemplate } = offer;
+    const { instance, installation } = idToOffer.get(id);
+    if (!instance || !installation) {
+      return;
+    }
     // We could get the instanceHandle and installationHandle from the
     // board and store them to prevent having to make this call each
     // time, but if we want the offers to be able to sent to the
     // frontend, we cannot store the instanceHandle and
     // installationHandle in these offer objects because the handles
     // are presences and we don't wish to send presences to the
     // frontend.
-    const instanceHandle = await E(board).getValue(instanceHandleBoardId);
-    const installationHandle = await E(board).getValue(
-      installationHandleBoardId,
-    );
-    const instance = display(instanceHandle);
-    const installation = display(installationHandle);
+    const instanceDisplay = display(instance);
+    const installationDisplay = display(installation);
     const alreadyDisplayed =
       inboxState.has(id) && inboxState.get(id).proposalForDisplay;
 
     const offerForDisplay = {
       ...offer,
-      instancePetname: instance.petname,
-      installationPetname: installation.petname,
+      actions: undefined,
+      installation: undefined,
+      instance: undefined,
```

Comment:
hmm don't we have these already? why undefined?

Reply:
We have them, but we don't want to send presences to the frontend, so we remove them.